src/HealthChecks.UI/ServiceCollectionExtensions.cs

```diff
@@ -151,5 +151,19 @@ await db.Configurations
                 }
             }
         }
+
+        private static volatile bool isDatabaseDeleted;
+        private static SemaphoreSlim deletionSemaphore = new SemaphoreSlim(1);
+        private static async Task EnsureMigratedAsync(this HealthChecksDb db)
+        {
+            await deletionSemaphore.WaitAsync();
+            if (!isDatabaseDeleted)
+            {
+                await db.Database.EnsureDeletedAsync();
+                isDatabaseDeleted = true;
+            }
+            await db.Database.MigrateAsync();

```

Comment:
Do you want all subsequent threads to execute this line?

Reply:
Not really. I read again the code and it is clear what I'm doing now. I think with the latest code I just pushed it will be clear. Thanks!